contracts/external/policies/UsageLimitPolicy.sol

```diff
@@ -0,0 +1,101 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.23;
+
+import "contracts/interfaces/IPolicy.sol";
+
+/**
+ * @title UsageLimitPolicy
+ * @notice A policy that limits the total amount of actions/userOps that can be performed within a permission.
+ */
+contract UsageLimitPolicy is IUserOpPolicy, IActionPolicy {
```

Comment:
why is this not a `I1271Policy`?

Reply:
because 1271 policy can not alter state, and usage limit should alter it